crates/sui-graphql-rpc/src/types/tx_digest.rs

```diff
@@ -0,0 +1,170 @@
+// Copyright (c) Mysten Labs, Inc.
+// SPDX-License-Identifier: Apache-2.0
+
+use async_graphql::*;
+use fastcrypto::encoding::{Base58, Encoding};
+use serde::de::Deserializer;
+use serde::{Deserialize, Serialize, Serializer};
+use serde_with::{serde_as, Bytes, DeserializeAs, SerializeAs};
+use std::fmt;
+use std::marker::PhantomData;
+
+const SUI_TRANSACTION_DIGEST_LENGTH: usize = 32;
+
+#[serde_as]
+#[derive(Serialize, Deserialize, Clone, Eq, PartialEq, Ord, PartialOrd, Hash, Copy)]
+pub(crate) struct TransactionDigest(
+    #[serde_as(as = "Readable<Base58, Bytes>")] [u8; SUI_TRANSACTION_DIGEST_LENGTH],
+);
+scalar!(TransactionDigest, "TransactionDigest");
```

Comment:
It's fine to have new helper types in the implementation, but we don't want to introduce new scalar types -- is there a way to avoid this affecting the external schema?